### PR TITLE
Move ApodiniObserveMetricsPrometheus to its own Swift package

### PR DESCRIPTION
**Pull request by moritzsternemann**
# Move ApodiniObserveMetricsPrometheus to its own Swift package

## :recycle: Current situation & Problem
The `SwiftPrometheus` dependency is fetched even if it is not used by the project.

## :bulb: Proposed solution
In the spirit of #235, `ApodiniObserveMetricsPrometheus` has been moved from being a target in the main Apodini repository to its own Swift package [Apodini/ApodiniObservePrometheus](https://github.com/Apodini/ApodiniObservePrometheus).

## :gear: Release Notes 
*Add a short summary of the feature as well as possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented if it appends or changes the public interface.*

## :heavy_plus_sign: Additional Information
To enable unit testing, the `XCTApodini` target has been exposed as a library. Additionally, the metrics test utils were moved to their own library/target named `XCTApodiniObserve`.

Eventually this should happen to more targets and `ApodiniObserve` is next on the list. However, due to ease of development of tracing functionality in `ApodiniObserve`, this is postponed for now.

### Testing
No changed functionality - existing tests were copied to the new repos.

ApodiniObservePrometheus' macOS tests are currently blocked by #374 

ApodiniObservePrometheus' linux release tests fail because the internal Swift module `SwiftShims` is missing, see [this run for example](https://github.com/Apodini/ApodiniObservePrometheus/runs/4341523454?check_suite_focus=true). Apparently this is caused by caching of the `.build` folder (https://stackoverflow.com/a/52426685/2657815), and GitHub actions currently doesn't have a feature to clean the cache manually. One suggested solution is to "abuse" a secret variable for a *cache version* that could be increased in such cases: https://github.community/t/how-to-clear-cache-in-github-actions/129038/5.

### Reviewer Nudging
Please make sure to check the [new Swift package Apodini/ApodiniObservePrometheus](https://github.com/Apodini/ApodiniObservePrometheus) Package.swift file and directory structure for any apparent issues.

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -27,3 +27,4 @@ Apodini contributors
 * [Sadik Ekin Özbay](https://github.com/sadikekin)
 * [Magdalena Papagianni](https://github.com/MagdalenaPap)
 * [Hans Santoso](https://github.com/hansfilbert)
+* [Moritz Sternemann](https://github.com/moritzsternemann)
```
Comment (PSchmiedmayer) [inline CONTRIBUTORS.md:30]: 🥳

File: Package.swift [hunk 2]
```diff
@@ -50,10 +50,13 @@ let package = Package(
         // Observe
         .library(name: "ApodiniObserve", targets: ["ApodiniObserve"]),
         .library(name: "ApodiniLoggingSupport", targets: ["ApodiniLoggingSupport"]),
-        .library(name: "ApodiniObserveMetricsPrometheus", targets: ["ApodiniObserveMetricsPrometheus"]),
+        .library(name: "XCTApodiniObserve", targets: ["XCTApodiniObserve"]),
```
Comment (PSchmiedmayer) [inline Package.swift]: Would suggest to move this line it next to XCTApodini in the Package.swift file a few lines below.

File: Sources/Apodini/Apodini.docc/Ecosystem/Observe.md [hunk 3]
```diff
@@ -265,15 +265,18 @@ struct ExampleWebService: WebService {
 }
 ```
 
-As the `MetricsConfiguration` is quite complex, `ApodiniObserve` offers a seperate `ApodiniObserveMetricsPrometheus` target that drastically simplifies the configuration process for the common [Prometheus](https://prometheus.io/) software stack. Include the dependency like:
+As the `MetricsConfiguration` is quite complex, `ApodiniObserve` offers a seperate [`ApodiniObservePrometheus` Swift package](https://github.com/Apodini/ApodiniObservePrometheus) that drastically simplifies the configuration process for the common [Prometheus](https://prometheus.io/) software stack. Include the dependency like:
 ```swift
+dependencies: [
+    .package(url: "https://github.com/Apodini/ApodiniObservePrometheus.git", from: "0.0.0"),
```
Comment (PSchmiedmayer) [inline Sources/Apodini/Apodini.docc/Ecosystem/Observe.md]: We should probably somehow remember to update this once we have tagged a release or somehow make this release independent.

File: TestWebService/Package.resolved [hunk 4]
```diff
@@ -73,6 +73,15 @@
           "version": "1.16.2"
         }
       },
+      {
```
Comment (PSchmiedmayer) [inline TestWebService/Package.resolved]: Can you check that the `Package.resolved` file is generated with the latest version of this commit? It looks strange that there are dependencies added when we move a target out of Apodini.

**Review by PSchmiedmayer — APPROVED** [review]
@moritzsternemann Thanks for the PR! Great next step to move ApodiniObserveMetricsPrometheus to its own Swift package.
I only had a few comments here and there which should be addressed before merging but nothing big that would require a new review 👍

**Comment by PSchmiedmayer** [comment]
@moritzsternemann One thing we should do after merging this PR and tagging a new `0.6.0` release and tagging a release in https://github.com/Apodini/ApodiniObservePrometheus is updating the validation of @philippzagar 's work (https://github.com/fa21-collaborative-drone-interactions/ScienceLabWebservice) to the latest version of both frameworks as this is currently relying on the fact that ApodiniObserveMetricsPrometheus is part of this repo here: https://github.com/fa21-collaborative-drone-interactions/ScienceLabWebservice/blob/0f67f56577763093caeb35a8d866cd82d8af6ed7/WebService/Package.swift#L32

I have provided you access to the https://github.com/fa21-collaborative-drone-interactions/ScienceLabWebservice repo.

**Comment by philippzagar** [comment]
Very nice to see that the `ApodiniObserveMetricsPrometheus` target is moving to a separate repository, that would have been actually a left task for my thesis, thanks! 👍 
As mentioned by Paul, it would be great if you would also adapt the [ScienceLabWebservice](https://github.com/fa21-collaborative-drone-interactions/ScienceLabWebservice) repository to the changed structure.

**Review by philippzagar — APPROVED** [review]

**Pull request merged**